pkg/sql/pgwire/hba_conf.go

```diff
@@ -247,17 +279,27 @@ func (s *Server) GetAuthenticationConfiguration() *hba.Conf {
 // RegisterAuthMethod registers an AuthMethod for pgwire
 // authentication and for use in HBA configuration.
 //
+// The minReqVersion, if provided and different from AnyVersion, is
+// called upon configuration to verify whether the current active
+// cluster version is at least the version specified.
+//
 // The checkEntry method, if provided, is called upon configuration
 // the cluster setting in the SQL client which attempts to change the
 // configuration. It can block the configuration if e.g. the syntax is
 // invalid.
-func RegisterAuthMethod(method string, fn AuthMethod, checkEntry CheckHBAEntry) {
-	hbaAuthMethods[method] = fn
+func RegisterAuthMethod(
+	method string, fn AuthMethod, minReqVersion cluster.VersionKey, checkEntry CheckHBAEntry,
+) {
+	hbaAuthMethods[method] = authMethodEntry{minReqVersion, fn}
 	if checkEntry != nil {
 		hbaCheckHBAEntries[method] = checkEntry
 	}
 }
 
+// AnyVersion can be passed as minReqVersion to RegisterAuthMethod()
+// to indicate there is no constraint on the cluster version.
+const AnyVersion cluster.VersionKey = -1
```

Comment:
I have a medium preference to remove this and force all auth methods to specify a real version. Can we use whatever the 19.2.0 release version was without breaking anybody? This would make it harder to misuse this feature (i.e., if someone copy-pasted an auth method with AnyVersion in it) and remove the need for special code that handles AnyVersion. Again, only medium preference, so if you feel that keeping it is better, that's fine too.